ts/screens/authentication/cie/components/CieConsentDataUsageRenderComponent.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { ElementType } from "react";
+import WebView, { WebViewNavigation } from "react-native-webview";
+import { WebViewHttpErrorEvent } from "react-native-webview/lib/WebViewTypes";
+import UnlockAccessScreen from "../../../onboarding/UnlockAccessScreen";
+import AuthErrorScreen from "./AuthErrorScreen";
+
+type ContentComponentProps = {
+  isLoginSuccess?: boolean;
+  LoaderComponent: ElementType;
+  hasError: boolean;
+  errorCode?: string;
+  cieConsentUri: string;
+  originSchemasWhiteList: Array<string>;
+  handleShouldStartLoading: (event: WebViewNavigation) => boolean;
+  handleWebViewError: () => void;
+  handleHttpError: (event: WebViewHttpErrorEvent) => void;
+  onRetry: () => void;
+  onCancel: () => void;
+};
+
+const CieConsentDataUsageRenderComponent = ({
+  isLoginSuccess,
+  LoaderComponent,
+  hasError,
+  errorCode,
+  cieConsentUri,
+  originSchemasWhiteList,
+  handleShouldStartLoading,
+  handleWebViewError,
+  handleHttpError,
+  onRetry,
+  onCancel
+}: ContentComponentProps) => {
+  if (isLoginSuccess) {
+    return <LoaderComponent />;
+  }
+  if (hasError) {
+    if (errorCode === "1002") {
+      return <UnlockAccessScreen authLevel="L2" />;
+    } else {
+      return (
+        <AuthErrorScreen
+          errorCode={errorCode}
+          onRetry={onRetry}
+          onCancel={onCancel}
+        />
+      );
```

Comment:
What about extracting those screens and give them life inside a navigator like we did with `CieWrongCiePinScreen`, `CieUnexpectedErrorScreen`, `CieExpiredOrInvalidScreen` and presenting them modally? This way we decouple all those screen and in the future we could navigate here also from other screen if we need too. What do you think?